content/blog/posts/esp-rust-15-07-2022.md

```diff
@@ -0,0 +1,125 @@
+
+### wokwi-server
+
+[`wokwi`](https://wokwi.com/) is an online simulator for IoT projects, it supports many chips including most Espressif ones, the `esp32`, `esp32s2` & `esp32c3`. By default, Wokwi provides an Arduino like IDE in the browser which handles compilation, and then runs the program in the simulator. What's really cool is that Wokwi also exposes a GDB stub which can be used to debug applications, a task which can be very tricky to setup on real hardware.
+
+Rust support (compilation in the browser) is on-going but not ready for general use just yet, however there is another way to use Wokwi for Rust development. Introducing [`wokwi-server`](https://github.com/MabezDev/wokwi-server), a way to build an application locally then run it in the browser - no hardware required! See the [cargo runner section](https://github.com/MabezDev/wokwi-server#as-a-cargo-runner) of the README for details. Once setup, `cargo run` is all you need.
+
+### esp-backtrace, esp-println & esp-alloc
+
+[esp-println](https://github.com/esp-rs/esp-println) prvoides `print!` & `println!` macros for all Espressif chips currently supported, as well as a `log::Log` implementation too. It is a standalone crate with no dependencies (uses the ROM functions burned into the chip at the factory). [esp-backtrace](https://github.com/esp-rs/esp-backtrace) is a crate for generating backtraces _on the chip_, and supports all the printing methods esp-println does. We currently have support for generating Xtensa & RISC-V backtraces which covers all Espressif chips. Finally we also created [esp-alloc](https://github.com/esp-rs/esp-alloc) which provides a bare metal allocator for all Espressif chips.
+
+### Devcontainer support
+
+Both the esp-idf-template & the esp-template repositories now have `cargo-generate` options to configure Devcontainers. It's as simple as filling in the options when cloning the template with `cargo-generate` and you can have a functioning esp rust enviroment immediatly, with all the tooling ready to go!
+
+## esp rust board
+
+The [esp rust board](https://github.com/esp-rs/esp-rust-board) is now available for purchase! You can find purchase links on [the Espressif website](https://www.espressif.com/en/products/devkits), it is listed as `ESP32-C3-DevKit-RUST-1`.
+
+## Ferrous systems training
+
+Tickets to [the Ferrous training](https://ferrous-systems.com/blog/announce-esp-training/) based on the `esp32c3` are now available too, checkout the options [here](https://ferrous-systems.com/training/#package-espressif-beginner-training). Booking the training includes a `ESP32-C3-DevKit-RUST-1 board` as part of the package.
+
+## What's next?
+
+Continue the rapid progress on the bare metal multi-chip HAL, `esp-hal`, including the new `async` effort. `esp-idf` version 5.0 is nearly here, and has some breaking changes that we'll need to handle in `libc` andq our `esp-idf` crates.
```

Comment:
There is a small typo at the end of the sentence: 
> libc` **andq** our

Reply:
Thank you!